SEED-platform/seed
# Add column display names to the database

#### Any background context you want to provide?
The project historically has magically merged a list of database fields with fields that existed after the user mapped/merged data. To support saving the list settings before data are imported, there was a need to store a list of all the columns in the Columns table.

Mapping data is a class that is used to manage manipulating list of columns for mapping, merging, and various other operations. 

#### What's this PR do?
* A default list of columns will be generated for a new organization upon creation.
* Several helper methods were created to support getting lists of columns for mapping in the Column object.
* All the tests were updated to use the `create_organization` helper method to ensure that the default columns are populated
* Several methods of the MappingData class were deprecated (or moved to Column). MappingData can/should be deprecated soon.
* Migration to add columns to existing organization (if non-existent), and add display names and data types to existing columns.

#### How should this be manually tested?
Import data in a new database. Load existing database and double check migrations.

#### What are the relevant tickets?
#1191 #1172 

#### Screenshots (if appropriate)
@axelstudios -- is it 'easy' to make the list of mapping suggestions the display names?

![image](https://user-images.githubusercontent.com/1907354/39586282-f2ef5e0a-4eb3-11e8-88c1-aed4c302b031.png)

#### Definition of Done:
